src/T2DMap.cpp

```diff
@@ -4257,13 +4262,29 @@ void T2DMap::slot_setArea()
         arealist_combobox->addItem(qsl("%1 (%2)").arg(sortedAreaList.at(i), QString::number(areaId)), QString::number(areaId));
     }
 
-
-
     if (set_room_area_dialog->exec() == QDialog::Rejected) { // Don't proceed if "cancel" was pressed
         return;
     }
 
-    int newAreaId = arealist_combobox->itemData(arealist_combobox->currentIndex()).toInt();
+    auto newArea = arealist_combobox->itemData(arealist_combobox->currentIndex());
+    int  newAreaId;
+    if (newArea.isValid()) {
+        newAreaId = newArea.toInt();
+    } else if (sortedAreaList.contains(arealist_combobox->currentText().trimmed())) {
+        newAreaId = mpMap->mpRoomDB->getAreaNamesMap().key(arealist_combobox->currentText());
+    } else {
+        auto newAreaName = arealist_combobox->currentText();
+        newAreaId = mpMap->mpRoomDB->addArea(newAreaName);
+        if (newAreaId == 0) {
+            return; // This should not happen, but if it fails addArea underneath will log reason
+        }
```

Comment:
```suggestion
        if (!newAreaId) {
            // This should not happen, but if it fails TRoomDB::addArea(...) above will log a reason to the main console
            mpMap->postMessage(tr("[ ERROR ] - Unable to add \"%1\" as an area to the map.\n"
                                  "See the \"[MAP ERROR:]\" message for the reason.",
                                  // Intentional separator between argument
                                  "The '[MAP ERROR:]' text should be the same as that used for the translation of \"[MAP ERROR:]%1\n\" in the 'TMAP::logerror(...)' function.").arg(newAreaName));
            return;
        }
```
:astonished: I've just checked and the error message that is referred to does not use the same system/formatting/presentation style as the "OK" message does so I think an "ERROR" message inviting the user to see what that says is a little more user-friendly then just the other one being squirted at them... :grinning: